src/Devices/Base/SimulatedEvent.cs

```diff
@@ -0,0 +1,77 @@
+using System;
+using System.Timers;
+
+namespace Transportation.Demo.Devices.Base
+{
+    public class SimulatedEvent
+    {
+        private Timer eventTimer;
+        private double interval;
+        private double jitter; 
+        public delegate void eventDelegate();
+        private long eventCount = 0;
+
+        private eventDelegate timerfunction;
+
+        // constructor
+        public SimulatedEvent(double interval, double jitter, eventDelegate timerfunction)
+        {
+            // jitter must be a positive value
+            if (jitter < 0)
+                throw new ArgumentException("Jitter value must be >= 0");
+            else
+                this.jitter = jitter;
+
+            // interval must be at greater than or equal to the jitter value
+            if (interval < jitter)
+                throw new ArgumentException("interval must be >= jitter value");
+            else
+                this.interval = interval;
+
+            this.timerfunction = timerfunction;
+
+            eventTimer = new Timer(this.CalculateNextInterval());
```

Comment:
initialize the timer and call Start instead of giving it CalculateNextInterval here, so that we have one authority place that starts the timers.

Reply:
removing initialization of the timer interval when the timer object is created. Interval will only be set during the "start" method.